internal/store/pod.go

```diff
@@ -1071,7 +1073,7 @@ func waitingReason(cs v1.ContainerStatus, reason string) bool {
 	if cs.State.Waiting == nil {
 		return false
 	}
-	return cs.State.Waiting.Reason == reason
+	return cs.State.Waiting.Reason == reason || strings.HasPrefix(cs.State.Waiting.Reason, postStartHookError)
```

Comment:
hmm can you explain why this change is needed as well?

Reply:
because in `poststarthook` case,the reason varies, as described in https://github.com/kubernetes/kube-state-metrics/issues/849,so the waiting reason either equals to any of `{"ContainerCreating", "CrashLoopBackOff", "CreateContainerConfigError", "ErrImagePull", "ImagePullBackOff", "CreateContainerError", "InvalidImageName"}` or starts with `PostStartHookError`.

Reply:
I see now I think, so we want to just create a label `reason="PostStartHookError"` if any reason is prefixed with "PostStartHookError......", correct?

Reply:
exactly

Reply:
I think we could use this instead https://golang.org/pkg/strings/#Contains